src/services/resources-getter.js

```diff
@@ -168,23 +150,14 @@ function ResourcesGetter(model, opts, params) {
       });
     }
 
-    return query;
-  }
-
-  function exec(query) {
-    return new P((resolve, reject) => {
-      if (params.sort) {
-        handleSortParam(query);
-      }
+    if (params.sort) {
+      handleSortParam(jsonQuery);
+    }
 
-      if (!hasRelationshipFilter()) {
-        query
-          .limit(getLimit())
-          .skip(getSkip());
-      }
+    jsonQuery.push({ $skip: getSkip() });
+    jsonQuery.push({ $limit: getLimit() });
```

Comment:
There are a `skip` and a `limit` pushed here but this looks to be the base query of `count` function.
Are you sure it won't generate a a bad count?

Reply:
I moved it inside the perform to avoid any issue